src/net.cpp

```diff
@@ -1406,6 +1421,12 @@ void CConnman::ThreadSocketHandler()
     }
 }
 
+void CConnman::SetMaxConnections(int newMaxConnections)
+{
+    nMaxConnections = std::min(newMaxConnections, nAvailableFds);
```

Comment:
since newMaxConnections is signed, may want to enforce the minimum of `MAX_ADDNODE_CONNECTIONS` here too with `std::max()`;

Reply:
Good idea. I'm not sure why I made it signed though, so what if I change it to `unsigned int`?

Reply:
Oh, I remember. Because `MAX_ADDNODE_CONNECTIONS` is also signed.

Reply:
Ah, yeah that's suboptimal on its own I think... but let's not change that now.